packages/markdown/remark/src/remark-shiki.ts

```diff
@@ -30,9 +30,15 @@ export interface ShikiConfig {
 	wrap?: boolean | null;
 }
 
-const remarkShiki = async ({ langs = [], theme = 'github-dark', wrap = false }: ShikiConfig) => {
-	const highlighter = await getHighlighter({ theme });
+/** 
+ * getHighlighter() is the most expensive step of Shiki. Instead of calling it on every page,
+ * cache it here as much as possible. Make sure that your highlighters can be cached, state-free.
```

Comment:
Does `components/Code.astro` suffer from the same problem? Maybe nothing we can do about that one...

Reply:
Not yet, but I'd really like to use this to prototype some caching/memoization ideas